spices/SPICE-0001-import-graph-analyzer-api.adoc

```diff
@@ -0,0 +1,336 @@
+= Import Graph Anaylzer API
+
+* Proposal: xref:SPICE-0001-import-graph-analyzer-api.adoc[SPICE-0001]
+* Author: link:https://github.com/bioball[Dan Chao]
+* Status: TBD
+* Implemented in: TBD
+* Category: Tooling
+
+== Introduction
+
+Provide a way to produce a graph of imports required by Pkl modules.
+
+== Motivation
+
+Many build tools require that all inputs are known to them ahead of time, for cacheing purposes.
+
+For example, Gradle's `Task` API provides controls for file-based inputs. It uses these inputs to perform an _up-to-date check_. If the input files have not changed, it skips doing the actual work described by the task.
+
+In Pkl parlance, the inputs for any evaluation includes the dependent Pkl modules that are imported in the entrypoint program.
+Given the following program:
+
+.myFile.pkl
+[source,pkl]
+----
+import "otherFile.pkl"
+
+result = otherFile
+----
+
+The inputs for evaluation of `myFile.pkl` includes file `otherFile.pkl`.
+
+Currently, our link:https://pkl-lang.org/main/current/pkl-gradle[Gradle Plugin] requires _manual_ changes by developers to correctly configure their up-to-date checks.
+If their program adds another file for import, developers need to update the `transitiveModules` property to include this new file:
```

Comment:
I'd assume that `fileTree("baseDir") { include("**/*.pkl") }` is often good enough.

Reply:
That's fine for simple use-cases, but it's problematic in large enough repos, especially if a single module's output doesn't need to be re-computed if an unrelated Pkl file changes.